A word about BC policy (#129)
glpi-project/docdev: source/sourcecode.rst

In order to contribute to the source code, you will have to know a few things about Git and the development model we follow.

> Versionning

Reviewer: ```suggestion
Versioning
```

-----------

Version numbers follow the `x.y.z` nomenclature, where `x` is a major release, `y` is an intermediate release, and `z` is a bugfix release.

Backward compatibility
----------------------

> Wherever possible, bugfix releases should not make any non-backwards compatible changes to our source code, so a plugin that has been made compatible with a `10.0.0` release should therefore be compatible, barring exceptions, with all `10.0.x` versions. In the event that an incompatibility is introduced in a bugfix version, please let us know, so that we can correct the problem.

Reviewer: ```suggestion
Wherever possible, bugfix releases should not make any non-backwards compatible changes to our source code, so a plugin that has been made compatible with a `10.0.0` release should therefore be compatible, barring exceptions, with all `10.0.x` versions. In the event that an incompatibility is introduced in a bugfix version, please let us know so that we can correct the problem.
```

* `xx/bugfixes` contains the next minor release source code,
* you should not care about all other branches that may exists, they should have been deleted right now.

The `main` branch is where new features are added. This code is reputed as **non stable**.

The `x.y/bugfixes` branches is where bugs are fixed. This code is reputed as *stable*.

> Those branches are created when a new major or intermediate version is released. At the time I wrote these lines, latest stable version is `10.0` so the current bugfix branch is `10.0/bugfixes`. We do not maintain previous stable versions, unless when critical bug or security issues are found, so old bugfixes branches are likely to not change; while they are still existing.

Reviewer: ```suggestion
Those branches are created when a new major or intermediate version is released. At the time I wrote these lines, the latest stable version is `10.0` so the current bugfix branch is `10.0/bugfixes`. We do not maintain previous stable versions, unless when critical bug or security issues are found, so old bugfixes branches are likely to not change; while they are still existing.
```

> Those branches are created when a new major or intermediate version is released. At the time I wrote these lines, latest stable version is `10.0` so the current bugfix branch is `10.0/bugfixes`. We do not maintain previous stable versions, unless when critical bug or security issues are found, so old bugfixes branches are likely to not change; while they are still existing.

Reviewer 2: > We do not maintain previous stable versions, unless when critical bug or security issues are found,

I think we should tell that the exception is limited to the latest previous intermediate version.

Author: Would this be OK ?

```suggestion
Those branches are created when a new major or intermediate version is released. At the time I wrote these lines, the latest stable version is `10.0` so the current bugfix branch is `10.0/bugfixes`. We do not maintain previous stable versions, so old bugfixes branches are likely to not change; while they are still existing. In case we found a critical bug or a security issue, we may exceptionally apply patches to the latest previous stable branch.
```

Reviewer 2: yes, this is ok for me.
